virtcontainers/sandbox.go

```diff
@@ -970,7 +970,12 @@ func (s *Sandbox) stopVM() error {
 	}
 
 	s.Logger().Info("Stopping VM")
-	return s.hypervisor.stopSandbox()
+	if err := s.hypervisor.stopSandbox(); err != nil {
+		s.store.Delete()
```

Comment:
This does not seem right to cleanup the store when the `stopSandbox()` fails. I mean doing this from such low level function is not appropriate IMO.